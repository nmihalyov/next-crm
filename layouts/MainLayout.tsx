import Head from 'next/head';
import { Layout } from 'antd';

import Header from '../components/Header/Header';

const MainLayout: React.FC<{
  children: React.ReactNode,
  title?: string
}> = props => {
  const { children, title } = props;

  return (<>
    <Head>
      <title>{title ? `${title} | Next CRM` : 'Next CRM'}</title>
      <meta name="description" content="TypeScript React project on NextJS" />
      <link rel="icon" href="/favicon.ico" />
    </Head>
    <Header />
    <Layout.Content>
      <Layout>
        <div className="container">
          {children}
        </div>
      </Layout>
    </Layout.Content>
  </>);
};

export default MainLayout;