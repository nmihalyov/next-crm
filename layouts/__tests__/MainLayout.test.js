import MainLayout from '../MainLayout';

describe('MainLayout:', () => {
  const setUp = props => shallow(<MainLayout {...props} />)
  const children = <div><p>Content</p></div>;

  it('renders MainLayout with default page title', () => {
    const component = setUp({children});

    expect(component).toMatchSnapshot();
  });

  it('renders MainLayout with custom page title', () => {
    const component = setUp({children, title: 'Main'});

    expect(component).toMatchSnapshot();
  });
});