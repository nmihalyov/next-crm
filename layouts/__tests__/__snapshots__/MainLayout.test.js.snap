// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout: renders MainLayout with custom page title 1`] = `
<Fragment>
  <Head>
    <title>
      Main | Next CRM
    </title>
    <meta
      content="TypeScript React project on NextJS"
      name="description"
    />
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <Memo(Header) />
  <Content>
    <Layout>
      <div
        className="container"
      >
        <div>
          <p>
            Content
          </p>
        </div>
      </div>
    </Layout>
  </Content>
</Fragment>
`;

exports[`MainLayout: renders MainLayout with default page title 1`] = `
<Fragment>
  <Head>
    <title>
      Next CRM
    </title>
    <meta
      content="TypeScript React project on NextJS"
      name="description"
    />
    <link
      href="/favicon.ico"
      rel="icon"
    />
  </Head>
  <Memo(Header) />
  <Content>
    <Layout>
      <div
        className="container"
      >
        <div>
          <p>
            Content
          </p>
        </div>
      </div>
    </Layout>
  </Content>
</Fragment>
`;
